src/layout/rect/iter.rs

```diff
@@ -35,6 +35,19 @@ impl Iterator for Rows {
         self.current_row_fwd += 1;
         Some(row)
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let remaining_count = self
+            .rect
+            .height
+            // Subtract the number of rows that were provided from the start of the `Rect`, from the
+            // total height.
+            .saturating_sub(self.current_row_fwd.saturating_sub(self.rect.y))
+            // Subtract the number of rows that were provided from the end of the `Rect`, from the
+            // total height.
+            .saturating_sub(self.rect.bottom().saturating_sub(self.current_row_back));
+        (remaining_count as usize, None)
```

Comment:
nit: variable names communicate the same thing as the comments pretty well here (and below)
Should the second element of the return value be `Some(count)` here? I think I'm reading https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint right.

```suggestion
        let start_count = self.current_row_fwd.saturating_sub(self.rect.top());
        let end_count = self.rect.bottom().saturating_sub(self.current_row_back);
        let count = self
            .rect
            .height
            .saturating_sub(start_count)
            .saturating_sub(end_count) as usize;
        (count, Some(count))
```

Reply:
You are correct, on returning `Some(count)`.  Will clear up the comments as well.